src/main/scala/bank/persistence/CustomerProfileDAO.scala

```diff
@@ -0,0 +1,7 @@
+package bank.persistence
+
+trait CustomerProfileDAO {
+  def getCustomerProfileByUsername(username: String): Option[CustomerProfile]
```

Comment:
any call to IO (network, disk, file etc) is blocking and could take an unexpected amount of time to complete. Consider wrapping the entire operation in Future